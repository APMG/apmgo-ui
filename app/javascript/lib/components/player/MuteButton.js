// @flow
import * as React from 'react'
import { connect } from 'react-redux'
import { muteClick, unmuteClick } from '../../redux/audio-player'

type MuteButtonProps = {
  muted: boolean,
  mute: () => {},
  unmute: () => {}
}

const MuteButtonPresenter = (props: MuteButtonProps) => {
  if (props.muted) {
    return (
      <button id="unmute-button" onClick={props.unmute}>Unmute</button>
    )
  } else {
    return (
      <button id="mute-button" onClick={props.mute}>Mute</button>
    )
  }
}

export default MuteButtonPresenter
