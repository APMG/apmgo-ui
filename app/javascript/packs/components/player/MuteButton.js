// @flow
import * as React from 'react'
import { dispatch, connect } from 'react-redux'
import { muteClick, unmuteClick } from '../../redux/audio-player'

type MuteButtonProps = {
  muted: boolean,
  mute: () => {},
  unmute: () => {}
}

export class MuteButtonPresenter extends React.Component {

  props: MuteButtonProps

  render() {
    if (this.props.muted) {
      return (
        <button id="unmute-button" onClick={this.props.unmute}>Unmute</button>
      )
    } else {
      return (
        <button id="mute-button" onClick={this.props.mute}>Mute</button>
      )
    }
  }
}

export const mapStateToProps = (newState: any) => {
  return {
    muted: newState.audioPlayer.muted
  }
}

export const mapDispatchToProps = (dispatch: dispatch) => {
  return {
    mute: () => {
      dispatch(muteClick())
    },
    unmute: () => {
      dispatch(unmuteClick())
    }
  }
}

export default MuteButtonPresenter //connect(
//   mapStateToProps,
//   mapDispatchToProps
// )(MuteButtonPresenter)
