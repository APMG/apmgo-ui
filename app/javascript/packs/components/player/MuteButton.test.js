
import React from 'react'
import getSnapshotJson from '../../redux/__tests__/mock-initial-state'
import MuteButton, { MuteButtonPresenter } from './MuteButton'

describe('Mute Button Component', () => {
  describe('Presenter', () => {
    it('Displays a Mute Button When Unmuted', () => {
      let tree = getSnapshotJson(<MuteButtonPresenter muted={false}/>)
      expect(tree).toMatchSnapshot()
    })
    it('Displays an Unmute Button When Muted', () => {
      let tree = getSnapshotJson(<MuteButtonPresenter muted={true}/>)
      expect(tree).toMatchSnapshot()
    })
  })
})
