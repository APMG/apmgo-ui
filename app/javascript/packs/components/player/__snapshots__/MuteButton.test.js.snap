// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mute Button Component Presenter Displays a Mute Button When Unmuted 1`] = `
<button
  onClick={undefined}
>
  Mute
</button>
`;

exports[`Mute Button Component Presenter Displays an Unmute Button When Muted 1`] = `
<button
  onClick={undefined}
>
  Unmute
</button>
`;
